### ISSUE TITLE
[Crash] Latest bleed, NotImplementedException, Arch Linux

### ISSUE THREAD

**Issue by ghost** [OpenRA/OpenRA#14504]
```
Exception of type `System.NotImplementedException`: The method or operation is not implemented.
  at System.IO.Stream.Write (System.ReadOnlySpan`1[T] source) [0x00000] in <7403a1ce1bb645ddb8f0aa7250aed945>:0 
  at OpenRA.Network.EchoConnection.Send (System.Int32 frame, System.Collections.Generic.List`1[T] orders) [0x00013] in <1cb3b40242f14d03b7a545dc92393682>:0 
  at OpenRA.Network.OrderManager.Tick () [0x00052] in <1cb3b40242f14d03b7a545dc92393682>:0 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x001ed] in <1cb3b40242f14d03b7a545dc92393682>:0 
  at OpenRA.Game.LogicTick () [0x00050] in <1cb3b40242f14d03b7a545dc92393682>:0 
  at OpenRA.Game.Loop () [0x000d6] in <1cb3b40242f14d03b7a545dc92393682>:0 
  at OpenRA.Game.Run () [0x00042] in <1cb3b40242f14d03b7a545dc92393682>:0 
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00011] in <1cb3b40242f14d03b7a545dc92393682>:0 
  at OpenRA.Program.Main (System.String[] args) [0x0004f] in <1cb3b40242f14d03b7a545dc92393682>:0 
```
```
Platform is Linux
Engine version is {DEV_VERSION}
Using SDL 2 with OpenGL renderer
Desktop resolution: 1280x1024
No custom resolution provided, using desktop resolution
Using resolution: 1280x1024
Using window scale 1.00
OpenGL version: 3.0 Mesa 17.2.5
Using default sound device
```
Trying to play RA
No mods work
Mono -V output:
```
Mono JIT compiler version 5.9.0.375 (master/c58ab6f Tue Dec  5 23:21:27 EET 2017)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       normal
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          supported, not enabled.
	GC:            sgen (concurrent by default)
```

I am very interested to know which change causes this

**Comment by Kosti1** [comment]
getting mono-alpha from aur, so ill probably close this issue if it works

**Comment by LipkeGu** [comment]
'''System.ReadOnlySpan`1[T]''' It seems that a Methode in Instanz 1 ist not implemented.

**Comment by pchote** [comment]
@Kosti1: Did downgrading mono fix the issue?

**Comment by shiningdracon** [comment]
I encountered the same crash, for both latest bleed and Release 20180218, build from source.
But the downloaded binary works fine. They are using the same mono.
```
mono -V
Mono JIT compiler version 5.10.0.140 (tarball Sat Feb 24 15:33:47 UTC 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          supported, not enabled.
	GC:            sgen (concurrent by default)
```
Seems the runtime environment is fine, but compile environment have problem.

Then, I checked the `System.IO.Stream.Write (System.ReadOnlySpan`1[T] source)` at
https://github.com/mono/mono/blob/ddf4e7be31bf884cfe11cabbfcaff4ca839d0277/mcs/class/corlib/corert/Stream.cs#L13

It is not implemented, but it works (in downloaded OpenRA binary)? So I guess, the binary compiled on Windows includes the implementation and could work for mono, the Linux compiled cannot.

**Comment by GraionDilach** [comment]
All binaries (including the Windows package) offered on the site are compiled on Travis, which is a Linux environment using mono 4.6.1 atm, though.

**Comment by shiningdracon** [comment]
That incomplete Stream.cs introduced on Oct 5, 2017, and the next release was 5.4.0.212, so versions before 5.4.0.212 works, make sense.

**Comment by pchote** [comment]
Weird.  The mono download page is showing 5.10 as the stable release, but the [changelog page](http://www.mono-project.com/docs/about-mono/releases/) shows 5.8 as the latest, with only a placeholder for 5.10.

**Comment by pchote** [comment]
It looks like we're going to need to do a fixup release ASAP to address this, the Windows compilation issues fixed by #14855, and the linux dependencies issues tracked in #14788.

**Comment by Mailaender** [comment]
https://github.com/OpenRA/OpenRA/commit/42f1db0d4b84f99bb7262af59607b55e2f180684
